doc/en/user/source/community/cog/cog.rst

```diff
@@ -0,0 +1,134 @@
+
+.. list-table::
+   :widths: 15 80
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Environment Variable
+     - Description
+   * - IIO_HTTP_MAX_REQUESTS
+     - The maximum number of requests to execute concurrently. Above this requests queue in memory, waiting for the running calls to complete. (Default 128)
+   * - IIO_HTTP_MAX_REQUESTS_PER_HOST
+     - The maximum number of requests for each host to execute concurrently. (Default 5)
+   * - IIO_HTTP_MAX_IDLE_CONNECTIONS
+     - The maximum number of idle connections. (Default 5)
+   * - IIO_HTTP_KEEP_ALIVE_TIME
+     - The Keep alive time (in seconds), representing maximum time that excess idle threads will wait for new tasks before terminating. (Default 60)
+
+AWS S3 Client configuration
+---------------------------
+A single S3 Asynchronous Client will be used for the same region and alias (url schema, i.e. http, https). 
+The following Environment Variables can be set to customize the pool for the asynchronous client for that particular alias. 
+On the table below, replace the "$ALIAS$" template with HTTP or HTTPS or S3 if you are configuring properties for these schema. 
+
+.. list-table::
+   :widths: 15 80
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Environment Variable
+     - Description
+   * - IIO_$ALIAS$_AWS_CORE_POOL_SIZE
+     - The core pool size for the S3 Client (Default 50)
+   * - IIO_$ALIAS$_AWS_MAX_POOL_SIZE
+     - The maximum number of thread to allow in the pool for the S3 Client (Default 128)
+   * - IIO_$ALIAS$_AWS_KEEP_ALIVE_TIME
+     - The Keep alive time (in seconds), representing maximum time that excess idle threads will wait for new tasks before terminating. (Default 10)
+   * - IIO_$ALIAS$_AWS_USER
+     - Default user (access key ID) for AWS basic authentication credentials
+   * - IIO_$ALIAS$_AWS_PASSWORD
+     - Default password (secret access key) for AWS basic authentication credentials
+
+Client configuration (System Properties)
+----------------------------------------
+Note that all the IIO  settings reported in the previous tables can also be specified using System Properties instead of Environment variables.
+You just need to replace UPPER CASE words with lower case words and underscores with dots.
+So, the value for Maximum HTTP requests can be specified by setting either a ``IIO_HTTP_MAX_REQUESTS`` Environment variable or a ``iio.http.max.requests`` Java System Property alternatively (Environment variables are checked first).
+
+By default, when accessing a COG, a first chunk of 16 KB is read in attempt to parset the header so that the reader will know offset and length of the available tiles. When dealing with files hosting many tiles, it might happen that the whole header won't fit in the first chunk. In this case additional reads (chunks of the same size) will be progressively made to complete loading the header.
```

Comment:
By default, when accessing a COG, an initial chunk of 16 KB is read in attempt to parse the header so that the reader will have the offset and length of the available tiles. When dealing with files hosting many tiles, it is possible that the whole header won't fit in the initial chunk. In this case additional reads (chunks of the same size) will be progressively made to complete loading the header.